workflow-dev/tpl/deis-registry-secret.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: registry-secret
+  namespace: deis
+  labels:
+    heritage: deis
+  annotations:
+    deis.io/objectstorage: "{{ or (env "REGISTRY_LOCATION") (.registry_location)}}"
+type: Opaque
+data: {{ if or (eq (env "REGISTRY_LOCATION") "gcs") (eq .registry_location "gcr")}}
+  key.json: {{ or (env "GCR_KEY_JSON") (.gcr.key_json) | b64enc}}
+  server: {{ or (env "GCR_SERVER") (.gcr.server) | b64enc}}{{ else if or (eq (env "REGISTRY_LOCATION") "ecr") (eq .registry_location "ecr")}}
+  accesskey: {{ or (env "ECR_ACCESS_KEY") (.ecr.accesskey) | b64enc }}
+  secretkey: {{ or (env "ECR_SECRET_KEY") (.ecr.secretkey) | b64enc }}
+  region: {{ or (env "ECR_REGION") (.ecr.region) | b64enc }}
+  registryid: {{ or (env "ECR_REGISTRY_ID") (.ecr.registryid) | b64enc }}
+  server: {{ or (env "ECR_SERVER") (.ecr.server) | b64enc }}{{else if or (eq (env "REGISTRY_LOCATION") "off-cluster") (eq .registry_location "off-cluster")}}
```

Comment:
I think we should use `hostname` here, even though the docker login command uses `--server`